src/trainer.py

```diff
@@ -1203,15 +1207,15 @@ def _save_target_train_checkpoints(
                 continue
             task_name = self.task_to_metric_mapping[metric_name]
             torch.save(
-                metric_info,
+                metric_states,
```

Comment:
Why are we saving this, but still iterating over self._metric_infos? This looks broken...

Reply:
If this code runs, then we should probably be able to just get rid of the loop.